tests/integration/dhcp_test.py

```diff
@@ -192,6 +193,126 @@ def test_dhcp_for_bond_with_ip_address_and_slave(dhcp_env,
     assertlib.assert_state(desired_state)
 
 
+def test_ipv4_dhcp_ignore_gateway(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv4']['enabled'] = True
+    dhcp_cli_desired_state['ipv4']['dhcp'] = True
+    dhcp_cli_desired_state['ipv4']['ignore-auto-gateway'] = True
+    dhcp_cli_desired_state['ipv6']['enabled'] = True
+
+    netapplier.apply(desired_state)
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support route query, need to make
```

Comment:
Assuming the dhcp server does provide a gateway, we can check it with a temporary helper (`ip route` output).

Reply:
This seems to be done